File: CPP/Program/Goblin.cpp
#pragma once
#include "Karakter.cpp"

class Goblin : public Karakter {
private:
    string FisikSpesial; // Atribut untuk menyimpan fisik spesial Goblin
public:
    // Konstruktor Goblin
    Goblin(string nama, int tinggi, int umur, Status s, string fisik)
        : Karakter(nama, tinggi, umur, s), FisikSpesial(fisik) {}

    // Setter & Getter untuk FisikSpesial
    void setFisikSpesial(string f) { FisikSpesial = f; }
    string getFisikSpesial() { return FisikSpesial; }

    // Override tampilkanInfo untuk menampilkan info Goblin
    void tampilkanInfo() override {
        cout << "[Goblin]" << endl;
        Karakter::tampilkanInfo();
        cout << "Fisik Spesial: " << FisikSpesial << endl;
        cout << "---------------------------" << endl;
    }
};

File: Python/Program/Goblin.py
from Karakter import Karakter

class Goblin(Karakter):
    def __init__(self, nama, tinggi, umur, status, fisik):
        super().__init__(nama, tinggi, umur, status)
        self._fisik_spesial = fisik  # Atribut fisik spesial Goblin

    # Setter & Getter
    def set_fisik_spesial(self, f): self._fisik_spesial = f
    def get_fisik_spesial(self): return self._fisik_spesial

    # Menampilkan info Goblin
    def tampilkan_info(self):
        print("[Goblin]")
        super().tampilkan_info()
        print(f"Fisik Spesial: {self._fisik_spesial}")
        print("---------------------------")
